lib/SILOptimizer/Mandatory/Differentiation.cpp

```diff
@@ -1292,6 +1550,26 @@ void DifferentiableActivityInfo::analyze(DominanceInfo *di,
   PROPAGATE_VARIED_FOR_STRUCT_EXTRACTION(StructElementAddr)
 #undef VISIT_STRUCT_ELEMENT_INNS
 
+        // Handle `br`.
+        else if (auto *bi = dyn_cast<BranchInst>(&inst)) {
```

Comment:
`TermInst` is all branch instructions' common base class. You can just call `getSuccessorBlocks()` to get all successors and assign activity to their arguments.

Reply:
I tried to use `TermInst` but couldn't figure out how to map branching operands to successor basic block arguments. This mapping differs between terminator instructions.

Let me know if you have further ideas.